### PR TITLE
Remove support for amtSol console

### PR DESCRIPTION
**Pull request by b10n1k**
No reference found in any test repo I looked in, neither in openQA.

https://progress.opensuse.org/issues/130417

### CODE REVIEW

**Comment by okurz** [comment]
Please ask previous contributors. And AFAICS the module was not declared as deprecated. We should only remove it after a reasonable deprecation period

**Comment by b10n1k** [comment]
> Please ask previous contributors. And AFAICS the module was not declared as deprecated. We should only remove it after a reasonable deprecation period

I see. I will do that

**Comment by b10n1k** [comment]
> > Please ask previous contributors. And AFAICS the module was not declared as deprecated. We should only remove it after a reasonable deprecation period
> 
> I see. I will do that. do you know how long is that period?

I found `./backend/amt.pm:34:    backend::baseclass::handle_deprecate_backend('AMT');`?!

**Comment by okurz** [comment]
> > > Please ask previous contributors. And AFAICS the module was not declared as deprecated. We should only remove it after a reasonable deprecation period
> > 
> > 
> > I see. I will do that. do you know how long is that period?
> 
> I found `./backend/amt.pm:34: backend::baseclass::handle_deprecate_backend('AMT');`?! for two years: [0918185](https://github.com/os-autoinst/os-autoinst/commit/0918185fc4071b64a4147763726932e6679e9112)

Right. That's enough. So we can just remove

**Comment by okurz** [comment]
Please rephrase the commit message s/Remove support for amtSol console/Remove obsolete amt backend/

and the failing coverage check is pointing you to another amt reference that needs to be removed:
https://app.codecov.io/gh/os-autoinst/os-autoinst/pull/2568?dropdown=coverage&src=pr&el=h1

File: t/23-baseclass.t [hunk 1]
```diff
@@ -892,4 +892,8 @@ subtest 'child process handling' => sub {
     combined_like { $baseclass->_stop_children_processes } qr/waitpid for $pid returned/, 'stopped child again';
 };
 
+subtest 'deprecate backend' => sub {
+    throws_ok { backend::baseclass::handle_deprecate_backend('AMT') } qr/is unsupported and planned to be\nremoved from os-autoinst eventually/, 'deprecated message is displayed';
```
Comment (kalikiana) [inline t/23-baseclass.t:896]: This is gone now, though? I'm confused why you're adding this test 🤔

Comment (b10n1k) [reply t/23-baseclass.t:896]: the code coverage was failing. Likely because uncover `handle_deprecate_backend` somehow

Comment (kalikiana) [reply t/23-baseclass.t:896]: Ah. I guess it was only used used with amt before so now it's not implicitly covered anymore. Makes sense!

**Review by kalikiana — APPROVED** [review]

**Comment by okurz** [comment]
latest codecov checks are missing but they had been there in before. I assume we are good and can merge now.

**Comment by b10n1k** [comment]
> latest codecov checks are missing but they had been there in before. I assume we are good and can merge now.

I was wondering why? if you know

**Comment by okurz** [comment]
> > latest codecov checks are missing but they had been there in before. I assume we are good and can merge now.
> 
> I was wondering why? if you know

I don't know specifically but we had similar problems with missing codecov reports in the past.

File: t/23-baseclass.t (hunk 1 above)
Comment (baierjan) [reply t/23-baseclass.t:896]: I guess this is related: https://app.codecov.io/gh/os-autoinst/os-autoinst/pull/2573/blob/backend/baseclass.pm#L1367

**Pull request merged**